jstests/ssl/x509/mkcert.py

```diff
@@ -520,17 +602,44 @@ def process_ecdsa_leaf(cert):
     reqargs = ['openssl', 'req', '-new', '-key', key, '-out', csr, '-subj', subject]
     x509args = ['openssl', 'x509', '-in', csr, '-out', pem, '-req', '-CA', ca, '-CAkey', ca, '-days', '7300', '-sha256', '-set_serial', serial]
     if mode == 'server':
-        reqargs.append('-reqexts=v3_req')
+        reqargs = reqargs + ['-reqexts', 'v3_req']
         extfile = tempfile.mkstemp()[1]
-        open(extfile, 'wt').write('subjectAltName=DNS:localhost,DNS:127.0.0.1')
+        with open(extfile, 'wt') as f:
+            f.write('basicConstraints=CA:FALSE\n')
+            f.write('subjectAltName=DNS:localhost,IP:127.0.0.1\n')
+            f.write('subjectKeyIdentifier=hash\n')
+            key_usage = ('keyUsage=nonRepudiation,digitalSignature,keyEncipherment\n'
+                        if (cert.get('tags') and "responder" in cert['tags'])
+                        else 'keyUsage=digitalSignature,keyEncipherment\n')
+            f.write(key_usage)
+            extended_key_usage = ('extendedKeyUsage=serverAuth,clientAuth,OCSPSigning\n'
+                                 if (cert.get('tags') and "responder" in cert['tags'])
+                                 else 'extendedKeyUsage=serverAuth,clientAuth\n')
+            f.write(extended_key_usage)
+            if cert.get('tags'):
+                if "ocsp" in cert['tags']:
+                    if not "responder" in cert['tags']:
+                        f.write('authorityInfoAccess = OCSP;URI:http://localhost:9001/power/level/,OCSP;URI:http://localhost:8100/status/\n')
+                        if "must-staple" in cert['tags']:
+                            f.write(f'{MUST_STAPLE_KEY_STR}={MUST_STAPLE_VALUE_STR}\n')
         x509args.append('-extfile')
         x509args.append(extfile)
 
     subprocess.check_call(['openssl', 'ecparam', '-name', 'prime256v1', '-genkey', '-out', key])
     subprocess.check_call(reqargs)
     subprocess.check_call(x509args)
 
-    open(dest, 'wt').write(get_header_comment(cert) + "\n" + open(pem, 'rt').read() + open(key, 'rt').read())
+    if cert.get('tags') and "responder" in cert['tags']:
```

Comment:
nit; You can write this as:   `if 'responder' in cert.get('tags', []):` to avoid some copypasta.

Reply:
Thanks for the tip! Nit addressed :)